// fastly/dist/model/ApexRedirectAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ApexRedirectAllOf model module.
 * @module model/ApexRedirectAllOf
 * @version 3.0.0-beta3
 */
var ApexRedirectAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ApexRedirectAllOf</code>.
   * @alias module:model/ApexRedirectAllOf
   */
  function ApexRedirectAllOf() {
    _classCallCheck(this, ApexRedirectAllOf);

    ApexRedirectAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ApexRedirectAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ApexRedirectAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ApexRedirectAllOf} obj Optional instance to populate.
     * @return {module:model/ApexRedirectAllOf} The populated <code>ApexRedirectAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ApexRedirectAllOf();

        if (data.hasOwnProperty('status_code')) {
          obj['status_code'] = _ApiClient["default"].convertToType(data['status_code'], 'Number');
        }

        if (data.hasOwnProperty('domains')) {
          obj['domains'] = _ApiClient["default"].convertToType(data['domains'], ['String']);
        }

        if (data.hasOwnProperty('feature_revision')) {
          obj['feature_revision'] = _ApiClient["default"].convertToType(data['feature_revision'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ApexRedirectAllOf;
}();
/**
 * HTTP status code used to redirect the client.
 * @member {module:model/ApexRedirectAllOf.StatusCodeEnum} status_code
 */


ApexRedirectAllOf.prototype['status_code'] = undefined;
/**
 * Array of apex domains that should redirect to their WWW subdomain.
 * @member {Array.<String>} domains
 */

ApexRedirectAllOf.prototype['domains'] = undefined;
/**
 * Revision number of the apex redirect feature implementation. Defaults to the most recent revision.
 * @member {Number} feature_revision
 */

ApexRedirectAllOf.prototype['feature_revision'] = undefined;
/**
 * Allowed values for the <code>status_code</code> property.
 * @enum {Number}
 * @readonly
 */

ApexRedirectAllOf['StatusCodeEnum'] = {
  /**
   * value: 301
   * @const
   */
  "301": 301,

  /**
   * value: 302
   * @const
   */
  "302": 302,

  /**
   * value: 307
   * @const
   */
  "307": 307,

  /**
   * value: 308
   * @const
   */
  "308": 308
};
var _default = ApexRedirectAllOf;
exports["default"] = _default;
