// fastly/dist/model/BillingAddressAttributes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BillingAddressAttributes model module.
 * @module model/BillingAddressAttributes
 * @version 3.0.0-beta3
 */
var BillingAddressAttributes = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BillingAddressAttributes</code>.
   * @alias module:model/BillingAddressAttributes
   */
  function BillingAddressAttributes() {
    _classCallCheck(this, BillingAddressAttributes);

    BillingAddressAttributes.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BillingAddressAttributes, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BillingAddressAttributes</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BillingAddressAttributes} obj Optional instance to populate.
     * @return {module:model/BillingAddressAttributes} The populated <code>BillingAddressAttributes</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BillingAddressAttributes();

        if (data.hasOwnProperty('address_1')) {
          obj['address_1'] = _ApiClient["default"].convertToType(data['address_1'], 'String');
        }

        if (data.hasOwnProperty('address_2')) {
          obj['address_2'] = _ApiClient["default"].convertToType(data['address_2'], 'String');
        }

        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }

        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }

        if (data.hasOwnProperty('locality')) {
          obj['locality'] = _ApiClient["default"].convertToType(data['locality'], 'String');
        }

        if (data.hasOwnProperty('postal_code')) {
          obj['postal_code'] = _ApiClient["default"].convertToType(data['postal_code'], 'String');
        }

        if (data.hasOwnProperty('state')) {
          obj['state'] = _ApiClient["default"].convertToType(data['state'], 'String');
        }

        if (data.hasOwnProperty('customer_id')) {
          obj['customer_id'] = _ApiClient["default"].convertToType(data['customer_id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return BillingAddressAttributes;
}();
/**
 * The first address line.
 * @member {String} address_1
 */


BillingAddressAttributes.prototype['address_1'] = undefined;
/**
 * The second address line.
 * @member {String} address_2
 */

BillingAddressAttributes.prototype['address_2'] = undefined;
/**
 * The city name.
 * @member {String} city
 */

BillingAddressAttributes.prototype['city'] = undefined;
/**
 * ISO 3166-1 two-letter country code.
 * @member {String} country
 */

BillingAddressAttributes.prototype['country'] = undefined;
/**
 * Other locality.
 * @member {String} locality
 */

BillingAddressAttributes.prototype['locality'] = undefined;
/**
 * Postal code (ZIP code for US addresses).
 * @member {String} postal_code
 */

BillingAddressAttributes.prototype['postal_code'] = undefined;
/**
 * The state or province name.
 * @member {String} state
 */

BillingAddressAttributes.prototype['state'] = undefined;
/**
 * @member {String} customer_id
 */

BillingAddressAttributes.prototype['customer_id'] = undefined;
var _default = BillingAddressAttributes;
exports["default"] = _default;
