// fastly/dist/model/BulkUpdateDictionaryItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BulkUpdateDictionaryItemAllOf = _interopRequireDefault(require("./BulkUpdateDictionaryItemAllOf"));

var _DictionaryItem = _interopRequireDefault(require("./DictionaryItem"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BulkUpdateDictionaryItem model module.
 * @module model/BulkUpdateDictionaryItem
 * @version 3.0.0-beta3
 */
var BulkUpdateDictionaryItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BulkUpdateDictionaryItem</code>.
   * @alias module:model/BulkUpdateDictionaryItem
   * @implements module:model/DictionaryItem
   * @implements module:model/BulkUpdateDictionaryItemAllOf
   */
  function BulkUpdateDictionaryItem() {
    _classCallCheck(this, BulkUpdateDictionaryItem);

    _DictionaryItem["default"].initialize(this);

    _BulkUpdateDictionaryItemAllOf["default"].initialize(this);

    BulkUpdateDictionaryItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BulkUpdateDictionaryItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BulkUpdateDictionaryItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BulkUpdateDictionaryItem} obj Optional instance to populate.
     * @return {module:model/BulkUpdateDictionaryItem} The populated <code>BulkUpdateDictionaryItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BulkUpdateDictionaryItem();

        _DictionaryItem["default"].constructFromObject(data, obj);

        _BulkUpdateDictionaryItemAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('item_key')) {
          obj['item_key'] = _ApiClient["default"].convertToType(data['item_key'], 'String');
        }

        if (data.hasOwnProperty('item_value')) {
          obj['item_value'] = _ApiClient["default"].convertToType(data['item_value'], 'String');
        }

        if (data.hasOwnProperty('op')) {
          obj['op'] = _ApiClient["default"].convertToType(data['op'], 'String');
        }
      }

      return obj;
    }
  }]);

  return BulkUpdateDictionaryItem;
}();
/**
 * Item key, maximum 256 characters.
 * @member {String} item_key
 */


BulkUpdateDictionaryItem.prototype['item_key'] = undefined;
/**
 * Item value, maximum 8000 characters.
 * @member {String} item_value
 */

BulkUpdateDictionaryItem.prototype['item_value'] = undefined;
/**
 * @member {module:model/BulkUpdateDictionaryItem.OpEnum} op
 */

BulkUpdateDictionaryItem.prototype['op'] = undefined; // Implement DictionaryItem interface:

/**
 * Item key, maximum 256 characters.
 * @member {String} item_key
 */

_DictionaryItem["default"].prototype['item_key'] = undefined;
/**
 * Item value, maximum 8000 characters.
 * @member {String} item_value
 */

_DictionaryItem["default"].prototype['item_value'] = undefined; // Implement BulkUpdateDictionaryItemAllOf interface:

/**
 * @member {module:model/BulkUpdateDictionaryItemAllOf.OpEnum} op
 */

_BulkUpdateDictionaryItemAllOf["default"].prototype['op'] = undefined;
/**
 * Allowed values for the <code>op</code> property.
 * @enum {String}
 * @readonly
 */

BulkUpdateDictionaryItem['OpEnum'] = {
  /**
   * value: "create"
   * @const
   */
  "create": "create",

  /**
   * value: "update"
   * @const
   */
  "update": "update",

  /**
   * value: "delete"
   * @const
   */
  "delete": "delete",

  /**
   * value: "upsert"
   * @const
   */
  "upsert": "upsert"
};
var _default = BulkUpdateDictionaryItem;
exports["default"] = _default;
