// fastly/dist/model/BulkUpdateAclEntryAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BulkUpdateAclEntryAllOf model module.
 * @module model/BulkUpdateAclEntryAllOf
 * @version 3.0.0-beta3
 */
var BulkUpdateAclEntryAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BulkUpdateAclEntryAllOf</code>.
   * @alias module:model/BulkUpdateAclEntryAllOf
   */
  function BulkUpdateAclEntryAllOf() {
    _classCallCheck(this, BulkUpdateAclEntryAllOf);

    BulkUpdateAclEntryAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BulkUpdateAclEntryAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BulkUpdateAclEntryAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BulkUpdateAclEntryAllOf} obj Optional instance to populate.
     * @return {module:model/BulkUpdateAclEntryAllOf} The populated <code>BulkUpdateAclEntryAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BulkUpdateAclEntryAllOf();

        if (data.hasOwnProperty('op')) {
          obj['op'] = _ApiClient["default"].convertToType(data['op'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return BulkUpdateAclEntryAllOf;
}();
/**
 * @member {module:model/BulkUpdateAclEntryAllOf.OpEnum} op
 */


BulkUpdateAclEntryAllOf.prototype['op'] = undefined;
/**
 * @member {String} id
 */

BulkUpdateAclEntryAllOf.prototype['id'] = undefined;
/**
 * Allowed values for the <code>op</code> property.
 * @enum {String}
 * @readonly
 */

BulkUpdateAclEntryAllOf['OpEnum'] = {
  /**
   * value: "create"
   * @const
   */
  "create": "create",

  /**
   * value: "update"
   * @const
   */
  "update": "update",

  /**
   * value: "delete"
   * @const
   */
  "delete": "delete"
};
var _default = BulkUpdateAclEntryAllOf;
exports["default"] = _default;
