// fastly/dist/model/Contact.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Contact model module.
 * @module model/Contact
 * @version 3.0.0-beta3
 */
var Contact = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Contact</code>.
   * @alias module:model/Contact
   */
  function Contact() {
    _classCallCheck(this, Contact);

    Contact.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Contact, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Contact</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Contact} obj Optional instance to populate.
     * @return {module:model/Contact} The populated <code>Contact</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Contact();

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'String');
        }

        if (data.hasOwnProperty('contact_type')) {
          obj['contact_type'] = _ApiClient["default"].convertToType(data['contact_type'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('phone')) {
          obj['phone'] = _ApiClient["default"].convertToType(data['phone'], 'String');
        }

        if (data.hasOwnProperty('customer_id')) {
          obj['customer_id'] = _ApiClient["default"].convertToType(data['customer_id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Contact;
}();
/**
 * The alphanumeric string representing the user for this customer contact.
 * @member {String} user_id
 */


Contact.prototype['user_id'] = undefined;
/**
 * The type of contact.
 * @member {module:model/Contact.ContactTypeEnum} contact_type
 */

Contact.prototype['contact_type'] = undefined;
/**
 * The name of this contact, when user_id is not provided.
 * @member {String} name
 */

Contact.prototype['name'] = undefined;
/**
 * The email of this contact, when a user_id is not provided.
 * @member {String} email
 */

Contact.prototype['email'] = undefined;
/**
 * The phone number for this contact. Required for primary, technical, and security contact types.
 * @member {String} phone
 */

Contact.prototype['phone'] = undefined;
/**
 * The alphanumeric string representing the customer for this customer contact.
 * @member {String} customer_id
 */

Contact.prototype['customer_id'] = undefined;
/**
 * Allowed values for the <code>contact_type</code> property.
 * @enum {String}
 * @readonly
 */

Contact['ContactTypeEnum'] = {
  /**
   * value: "primary"
   * @const
   */
  "primary": "primary",

  /**
   * value: "billing"
   * @const
   */
  "billing": "billing",

  /**
   * value: "technical"
   * @const
   */
  "technical": "technical",

  /**
   * value: "security"
   * @const
   */
  "security": "security",

  /**
   * value: "emergency"
   * @const
   */
  "emergency": "emergency",

  /**
   * value: "general compliance"
   * @const
   */
  "general compliance": "general compliance"
};
var _default = Contact;
exports["default"] = _default;
