// fastly/dist/model/HistoricalUsageResults.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The HistoricalUsageResults model module.
 * @module model/HistoricalUsageResults
 * @version 3.0.0-beta3
 */
var HistoricalUsageResults = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>HistoricalUsageResults</code>.
   * @alias module:model/HistoricalUsageResults
   */
  function HistoricalUsageResults() {
    _classCallCheck(this, HistoricalUsageResults);

    HistoricalUsageResults.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(HistoricalUsageResults, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>HistoricalUsageResults</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/HistoricalUsageResults} obj Optional instance to populate.
     * @return {module:model/HistoricalUsageResults} The populated <code>HistoricalUsageResults</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new HistoricalUsageResults();

        if (data.hasOwnProperty('bandwidth')) {
          obj['bandwidth'] = _ApiClient["default"].convertToType(data['bandwidth'], 'Number');
        }

        if (data.hasOwnProperty('requests')) {
          obj['requests'] = _ApiClient["default"].convertToType(data['requests'], 'Number');
        }

        if (data.hasOwnProperty('compute_requests')) {
          obj['compute_requests'] = _ApiClient["default"].convertToType(data['compute_requests'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return HistoricalUsageResults;
}();
/**
 * @member {Number} bandwidth
 */


HistoricalUsageResults.prototype['bandwidth'] = undefined;
/**
 * @member {Number} requests
 */

HistoricalUsageResults.prototype['requests'] = undefined;
/**
 * @member {Number} compute_requests
 */

HistoricalUsageResults.prototype['compute_requests'] = undefined;
var _default = HistoricalUsageResults;
exports["default"] = _default;
