// fastly/dist/model/IamPermission.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The IamPermission model module.
 * @module model/IamPermission
 * @version 3.0.0-beta3
 */
var IamPermission = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IamPermission</code>.
   * @alias module:model/IamPermission
   */
  function IamPermission() {
    _classCallCheck(this, IamPermission);

    IamPermission.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(IamPermission, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>IamPermission</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/IamPermission} obj Optional instance to populate.
     * @return {module:model/IamPermission} The populated <code>IamPermission</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IamPermission();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('object')) {
          obj['object'] = _ApiClient["default"].convertToType(data['object'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('resource_name')) {
          obj['resource_name'] = _ApiClient["default"].convertToType(data['resource_name'], 'String');
        }

        if (data.hasOwnProperty('resource_description')) {
          obj['resource_description'] = _ApiClient["default"].convertToType(data['resource_description'], 'String');
        }

        if (data.hasOwnProperty('scope')) {
          obj['scope'] = _ApiClient["default"].convertToType(data['scope'], 'String');
        }
      }

      return obj;
    }
  }]);

  return IamPermission;
}();
/**
 * Alphanumeric string identifying the permission.
 * @member {String} id
 */


IamPermission.prototype['id'] = undefined;
/**
 * The type of the object.
 * @member {String} object
 */

IamPermission.prototype['object'] = undefined;
/**
 * Name of the permission.
 * @member {String} name
 */

IamPermission.prototype['name'] = undefined;
/**
 * The description of the permission.
 * @member {String} description
 */

IamPermission.prototype['description'] = undefined;
/**
 * The name of the resource the operation will be performed on.
 * @member {String} resource_name
 */

IamPermission.prototype['resource_name'] = undefined;
/**
 * The description of the resource.
 * @member {String} resource_description
 */

IamPermission.prototype['resource_description'] = undefined;
/**
 * Permissions are either \"service\" level or \"account\" level.
 * @member {String} scope
 */

IamPermission.prototype['scope'] = undefined;
var _default = IamPermission;
exports["default"] = _default;
