// fastly/dist/model/IamRoleAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The IamRoleAllOf model module.
 * @module model/IamRoleAllOf
 * @version 3.0.0-beta3
 */
var IamRoleAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IamRoleAllOf</code>.
   * @alias module:model/IamRoleAllOf
   */
  function IamRoleAllOf() {
    _classCallCheck(this, IamRoleAllOf);

    IamRoleAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(IamRoleAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>IamRoleAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/IamRoleAllOf} obj Optional instance to populate.
     * @return {module:model/IamRoleAllOf} The populated <code>IamRoleAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IamRoleAllOf();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('object')) {
          obj['object'] = _ApiClient["default"].convertToType(data['object'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('custom')) {
          obj['custom'] = _ApiClient["default"].convertToType(data['custom'], 'Boolean');
        }

        if (data.hasOwnProperty('permissions_count')) {
          obj['permissions_count'] = _ApiClient["default"].convertToType(data['permissions_count'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return IamRoleAllOf;
}();
/**
 * Alphanumeric string identifying the role.
 * @member {String} id
 */


IamRoleAllOf.prototype['id'] = undefined;
/**
 * The type of the object.
 * @member {String} object
 */

IamRoleAllOf.prototype['object'] = undefined;
/**
 * Name of the role.
 * @member {String} name
 */

IamRoleAllOf.prototype['name'] = undefined;
/**
 * Description of the role.
 * @member {String} description
 */

IamRoleAllOf.prototype['description'] = undefined;
/**
 * This attribute is set to `true` if the role is managed by the customer. It is set to `false` if the role was created by Fastly.
 * @member {Boolean} custom
 */

IamRoleAllOf.prototype['custom'] = undefined;
/**
 * Number of permissions assigned to the role.
 * @member {Number} permissions_count
 */

IamRoleAllOf.prototype['permissions_count'] = undefined;
var _default = IamRoleAllOf;
exports["default"] = _default;
