// fastly/dist/model/IamServiceGroupAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The IamServiceGroupAllOf model module.
 * @module model/IamServiceGroupAllOf
 * @version 3.0.0-beta3
 */
var IamServiceGroupAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IamServiceGroupAllOf</code>.
   * @alias module:model/IamServiceGroupAllOf
   */
  function IamServiceGroupAllOf() {
    _classCallCheck(this, IamServiceGroupAllOf);

    IamServiceGroupAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(IamServiceGroupAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>IamServiceGroupAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/IamServiceGroupAllOf} obj Optional instance to populate.
     * @return {module:model/IamServiceGroupAllOf} The populated <code>IamServiceGroupAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IamServiceGroupAllOf();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('object')) {
          obj['object'] = _ApiClient["default"].convertToType(data['object'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('services_count')) {
          obj['services_count'] = _ApiClient["default"].convertToType(data['services_count'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return IamServiceGroupAllOf;
}();
/**
 * Alphanumeric string identifying the service group.
 * @member {String} id
 */


IamServiceGroupAllOf.prototype['id'] = undefined;
/**
 * The type of the object.
 * @member {String} object
 */

IamServiceGroupAllOf.prototype['object'] = undefined;
/**
 * Name of the service group.
 * @member {String} name
 */

IamServiceGroupAllOf.prototype['name'] = undefined;
/**
 * Description of the service group.
 * @member {String} description
 */

IamServiceGroupAllOf.prototype['description'] = undefined;
/**
 * Number of services in the service group.
 * @member {Number} services_count
 */

IamServiceGroupAllOf.prototype['services_count'] = undefined;
var _default = IamServiceGroupAllOf;
exports["default"] = _default;
