// fastly/dist/model/IamUserGroupAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The IamUserGroupAllOf model module.
 * @module model/IamUserGroupAllOf
 * @version 3.0.0-beta3
 */
var IamUserGroupAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IamUserGroupAllOf</code>.
   * @alias module:model/IamUserGroupAllOf
   */
  function IamUserGroupAllOf() {
    _classCallCheck(this, IamUserGroupAllOf);

    IamUserGroupAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(IamUserGroupAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>IamUserGroupAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/IamUserGroupAllOf} obj Optional instance to populate.
     * @return {module:model/IamUserGroupAllOf} The populated <code>IamUserGroupAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IamUserGroupAllOf();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('invitations_count')) {
          obj['invitations_count'] = _ApiClient["default"].convertToType(data['invitations_count'], 'Number');
        }

        if (data.hasOwnProperty('users_count')) {
          obj['users_count'] = _ApiClient["default"].convertToType(data['users_count'], 'Number');
        }

        if (data.hasOwnProperty('roles_count')) {
          obj['roles_count'] = _ApiClient["default"].convertToType(data['roles_count'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return IamUserGroupAllOf;
}();
/**
 * Alphanumeric string identifying the user group.
 * @member {String} id
 */


IamUserGroupAllOf.prototype['id'] = undefined;
/**
 * Name of the user group.
 * @member {String} name
 */

IamUserGroupAllOf.prototype['name'] = undefined;
/**
 * Description of the user group.
 * @member {String} description
 */

IamUserGroupAllOf.prototype['description'] = undefined;
/**
 * Number of invitations added to the user group.
 * @member {Number} invitations_count
 */

IamUserGroupAllOf.prototype['invitations_count'] = undefined;
/**
 * Number of users added to the user group.
 * @member {Number} users_count
 */

IamUserGroupAllOf.prototype['users_count'] = undefined;
/**
 * Number of roles added to the user group.
 * @member {Number} roles_count
 */

IamUserGroupAllOf.prototype['roles_count'] = undefined;
var _default = IamUserGroupAllOf;
exports["default"] = _default;
