// fastly/dist/model/PackageResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Package = _interopRequireDefault(require("./Package"));

var _PackageMetadata = _interopRequireDefault(require("./PackageMetadata"));

var _PackageResponseAllOf = _interopRequireDefault(require("./PackageResponseAllOf"));

var _ServiceIdAndVersion = _interopRequireDefault(require("./ServiceIdAndVersion"));

var _Timestamps = _interopRequireDefault(require("./Timestamps"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The PackageResponse model module.
 * @module model/PackageResponse
 * @version 3.0.0-beta3
 */
var PackageResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PackageResponse</code>.
   * @alias module:model/PackageResponse
   * @implements module:model/Package
   * @implements module:model/ServiceIdAndVersion
   * @implements module:model/Timestamps
   * @implements module:model/PackageResponseAllOf
   */
  function PackageResponse() {
    _classCallCheck(this, PackageResponse);

    _Package["default"].initialize(this);

    _ServiceIdAndVersion["default"].initialize(this);

    _Timestamps["default"].initialize(this);

    _PackageResponseAllOf["default"].initialize(this);

    PackageResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(PackageResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PackageResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PackageResponse} obj Optional instance to populate.
     * @return {module:model/PackageResponse} The populated <code>PackageResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PackageResponse();

        _Package["default"].constructFromObject(data, obj);

        _ServiceIdAndVersion["default"].constructFromObject(data, obj);

        _Timestamps["default"].constructFromObject(data, obj);

        _PackageResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _PackageMetadata["default"].constructFromObject(data['metadata']);
        }

        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }

        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return PackageResponse;
}();
/**
 * @member {module:model/PackageMetadata} metadata
 */


PackageResponse.prototype['metadata'] = undefined;
/**
 * @member {String} service_id
 */

PackageResponse.prototype['service_id'] = undefined;
/**
 * @member {Number} version
 */

PackageResponse.prototype['version'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */

PackageResponse.prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

PackageResponse.prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

PackageResponse.prototype['updated_at'] = undefined;
/**
 * Alphanumeric string identifying the package.
 * @member {String} id
 */

PackageResponse.prototype['id'] = undefined; // Implement Package interface:

/**
 * @member {module:model/PackageMetadata} metadata
 */

_Package["default"].prototype['metadata'] = undefined; // Implement ServiceIdAndVersion interface:

/**
 * @member {String} service_id
 */

_ServiceIdAndVersion["default"].prototype['service_id'] = undefined;
/**
 * @member {Number} version
 */

_ServiceIdAndVersion["default"].prototype['version'] = undefined; // Implement Timestamps interface:

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */

_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

_Timestamps["default"].prototype['updated_at'] = undefined; // Implement PackageResponseAllOf interface:

/**
 * Alphanumeric string identifying the package.
 * @member {String} id
 */

_PackageResponseAllOf["default"].prototype['id'] = undefined;
var _default = PackageResponse;
exports["default"] = _default;
