// fastly/dist/model/ResourceResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ResourceCreate = _interopRequireDefault(require("./ResourceCreate"));

var _ResourceResponseAllOf = _interopRequireDefault(require("./ResourceResponseAllOf"));

var _Timestamps = _interopRequireDefault(require("./Timestamps"));

var _TypeResource = _interopRequireDefault(require("./TypeResource"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ResourceResponse model module.
 * @module model/ResourceResponse
 * @version 3.0.0-beta3
 */
var ResourceResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ResourceResponse</code>.
   * @alias module:model/ResourceResponse
   * @implements module:model/Timestamps
   * @implements module:model/ResourceCreate
   * @implements module:model/ResourceResponseAllOf
   */
  function ResourceResponse() {
    _classCallCheck(this, ResourceResponse);

    _Timestamps["default"].initialize(this);

    _ResourceCreate["default"].initialize(this);

    _ResourceResponseAllOf["default"].initialize(this);

    ResourceResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ResourceResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ResourceResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ResourceResponse} obj Optional instance to populate.
     * @return {module:model/ResourceResponse} The populated <code>ResourceResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ResourceResponse();

        _Timestamps["default"].constructFromObject(data, obj);

        _ResourceCreate["default"].constructFromObject(data, obj);

        _ResourceResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }

        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('resource_id')) {
          obj['resource_id'] = _ApiClient["default"].convertToType(data['resource_id'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('href')) {
          obj['href'] = _ApiClient["default"].convertToType(data['href'], 'String');
        }

        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }

        if (data.hasOwnProperty('resource_type')) {
          obj['resource_type'] = _TypeResource["default"].constructFromObject(data['resource_type']);
        }
      }

      return obj;
    }
  }]);

  return ResourceResponse;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */


ResourceResponse.prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

ResourceResponse.prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

ResourceResponse.prototype['updated_at'] = undefined;
/**
 * The name of the resource.
 * @member {String} name
 */

ResourceResponse.prototype['name'] = undefined;
/**
 * The ID of the linked resource.
 * @member {String} resource_id
 */

ResourceResponse.prototype['resource_id'] = undefined;
/**
 * An alphanumeric string identifying the resource.
 * @member {String} id
 */

ResourceResponse.prototype['id'] = undefined;
/**
 * The path to the resource.
 * @member {String} href
 */

ResourceResponse.prototype['href'] = undefined;
/**
 * Alphanumeric string identifying the service.
 * @member {String} service_id
 */

ResourceResponse.prototype['service_id'] = undefined;
/**
 * Integer identifying a service version.
 * @member {Number} version
 */

ResourceResponse.prototype['version'] = undefined;
/**
 * @member {module:model/TypeResource} resource_type
 */

ResourceResponse.prototype['resource_type'] = undefined; // Implement Timestamps interface:

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */

_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

_Timestamps["default"].prototype['updated_at'] = undefined; // Implement ResourceCreate interface:

/**
 * The name of the resource.
 * @member {String} name
 */

_ResourceCreate["default"].prototype['name'] = undefined;
/**
 * The ID of the linked resource.
 * @member {String} resource_id
 */

_ResourceCreate["default"].prototype['resource_id'] = undefined; // Implement ResourceResponseAllOf interface:

/**
 * An alphanumeric string identifying the resource.
 * @member {String} id
 */

_ResourceResponseAllOf["default"].prototype['id'] = undefined;
/**
 * The path to the resource.
 * @member {String} href
 */

_ResourceResponseAllOf["default"].prototype['href'] = undefined;
/**
 * Alphanumeric string identifying the service.
 * @member {String} service_id
 */

_ResourceResponseAllOf["default"].prototype['service_id'] = undefined;
/**
 * Integer identifying a service version.
 * @member {Number} version
 */

_ResourceResponseAllOf["default"].prototype['version'] = undefined;
/**
 * @member {module:model/TypeResource} resource_type
 */

_ResourceResponseAllOf["default"].prototype['resource_type'] = undefined;
var _default = ResourceResponse;
exports["default"] = _default;
