// fastly/dist/model/ResourceResponseAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _TypeResource = _interopRequireDefault(require("./TypeResource"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ResourceResponseAllOf model module.
 * @module model/ResourceResponseAllOf
 * @version 3.0.0-beta3
 */
var ResourceResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ResourceResponseAllOf</code>.
   * @alias module:model/ResourceResponseAllOf
   */
  function ResourceResponseAllOf() {
    _classCallCheck(this, ResourceResponseAllOf);

    ResourceResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ResourceResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ResourceResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ResourceResponseAllOf} obj Optional instance to populate.
     * @return {module:model/ResourceResponseAllOf} The populated <code>ResourceResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ResourceResponseAllOf();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('href')) {
          obj['href'] = _ApiClient["default"].convertToType(data['href'], 'String');
        }

        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }

        if (data.hasOwnProperty('resource_type')) {
          obj['resource_type'] = _TypeResource["default"].constructFromObject(data['resource_type']);
        }
      }

      return obj;
    }
  }]);

  return ResourceResponseAllOf;
}();
/**
 * An alphanumeric string identifying the resource.
 * @member {String} id
 */


ResourceResponseAllOf.prototype['id'] = undefined;
/**
 * The path to the resource.
 * @member {String} href
 */

ResourceResponseAllOf.prototype['href'] = undefined;
/**
 * Alphanumeric string identifying the service.
 * @member {String} service_id
 */

ResourceResponseAllOf.prototype['service_id'] = undefined;
/**
 * Integer identifying a service version.
 * @member {Number} version
 */

ResourceResponseAllOf.prototype['version'] = undefined;
/**
 * @member {module:model/TypeResource} resource_type
 */

ResourceResponseAllOf.prototype['resource_type'] = undefined;
var _default = ResourceResponseAllOf;
exports["default"] = _default;
