// fastly/dist/model/ServiceVersionDetailOrNull.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BackendResponse = _interopRequireDefault(require("./BackendResponse"));

var _CacheSettingResponse = _interopRequireDefault(require("./CacheSettingResponse"));

var _ConditionResponse = _interopRequireDefault(require("./ConditionResponse"));

var _Director = _interopRequireDefault(require("./Director"));

var _DomainResponse = _interopRequireDefault(require("./DomainResponse"));

var _GzipResponse = _interopRequireDefault(require("./GzipResponse"));

var _HeaderResponse = _interopRequireDefault(require("./HeaderResponse"));

var _HealthcheckResponse = _interopRequireDefault(require("./HealthcheckResponse"));

var _RequestSettingsResponse = _interopRequireDefault(require("./RequestSettingsResponse"));

var _ResponseObjectResponse = _interopRequireDefault(require("./ResponseObjectResponse"));

var _SchemasSnippetResponse = _interopRequireDefault(require("./SchemasSnippetResponse"));

var _SchemasVclResponse = _interopRequireDefault(require("./SchemasVclResponse"));

var _ServiceVersionDetail = _interopRequireDefault(require("./ServiceVersionDetail"));

var _Settings = _interopRequireDefault(require("./Settings"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ServiceVersionDetailOrNull model module.
 * @module model/ServiceVersionDetailOrNull
 * @version 3.0.0-beta3
 */
var ServiceVersionDetailOrNull = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ServiceVersionDetailOrNull</code>.
   * @alias module:model/ServiceVersionDetailOrNull
   * @implements module:model/ServiceVersionDetail
   */
  function ServiceVersionDetailOrNull() {
    _classCallCheck(this, ServiceVersionDetailOrNull);

    _ServiceVersionDetail["default"].initialize(this);

    ServiceVersionDetailOrNull.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ServiceVersionDetailOrNull, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ServiceVersionDetailOrNull</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ServiceVersionDetailOrNull} obj Optional instance to populate.
     * @return {module:model/ServiceVersionDetailOrNull} The populated <code>ServiceVersionDetailOrNull</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ServiceVersionDetailOrNull();

        _ServiceVersionDetail["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('active')) {
          obj['active'] = _ApiClient["default"].convertToType(data['active'], 'Boolean');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('deployed')) {
          obj['deployed'] = _ApiClient["default"].convertToType(data['deployed'], 'Boolean');
        }

        if (data.hasOwnProperty('locked')) {
          obj['locked'] = _ApiClient["default"].convertToType(data['locked'], 'Boolean');
        }

        if (data.hasOwnProperty('number')) {
          obj['number'] = _ApiClient["default"].convertToType(data['number'], 'Number');
        }

        if (data.hasOwnProperty('staging')) {
          obj['staging'] = _ApiClient["default"].convertToType(data['staging'], 'Boolean');
        }

        if (data.hasOwnProperty('testing')) {
          obj['testing'] = _ApiClient["default"].convertToType(data['testing'], 'Boolean');
        }

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }

        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }

        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }

        if (data.hasOwnProperty('backends')) {
          obj['backends'] = _ApiClient["default"].convertToType(data['backends'], [_BackendResponse["default"]]);
        }

        if (data.hasOwnProperty('cache_settings')) {
          obj['cache_settings'] = _ApiClient["default"].convertToType(data['cache_settings'], [_CacheSettingResponse["default"]]);
        }

        if (data.hasOwnProperty('conditions')) {
          obj['conditions'] = _ApiClient["default"].convertToType(data['conditions'], [_ConditionResponse["default"]]);
        }

        if (data.hasOwnProperty('directors')) {
          obj['directors'] = _ApiClient["default"].convertToType(data['directors'], [_Director["default"]]);
        }

        if (data.hasOwnProperty('domains')) {
          obj['domains'] = _ApiClient["default"].convertToType(data['domains'], [_DomainResponse["default"]]);
        }

        if (data.hasOwnProperty('gzips')) {
          obj['gzips'] = _ApiClient["default"].convertToType(data['gzips'], [_GzipResponse["default"]]);
        }

        if (data.hasOwnProperty('headers')) {
          obj['headers'] = _ApiClient["default"].convertToType(data['headers'], [_HeaderResponse["default"]]);
        }

        if (data.hasOwnProperty('healthchecks')) {
          obj['healthchecks'] = _ApiClient["default"].convertToType(data['healthchecks'], [_HealthcheckResponse["default"]]);
        }

        if (data.hasOwnProperty('request_settings')) {
          obj['request_settings'] = _ApiClient["default"].convertToType(data['request_settings'], [_RequestSettingsResponse["default"]]);
        }

        if (data.hasOwnProperty('response_objects')) {
          obj['response_objects'] = _ApiClient["default"].convertToType(data['response_objects'], [_ResponseObjectResponse["default"]]);
        }

        if (data.hasOwnProperty('settings')) {
          obj['settings'] = _ApiClient["default"].convertToType(data['settings'], _Settings["default"]);
        }

        if (data.hasOwnProperty('snippets')) {
          obj['snippets'] = _ApiClient["default"].convertToType(data['snippets'], [_SchemasSnippetResponse["default"]]);
        }

        if (data.hasOwnProperty('vcls')) {
          obj['vcls'] = _ApiClient["default"].convertToType(data['vcls'], [_SchemasVclResponse["default"]]);
        }

        if (data.hasOwnProperty('wordpress')) {
          obj['wordpress'] = _ApiClient["default"].convertToType(data['wordpress'], [Object]);
        }
      }

      return obj;
    }
  }]);

  return ServiceVersionDetailOrNull;
}();
/**
 * Whether this is the active version or not.
 * @member {Boolean} active
 * @default false
 */


ServiceVersionDetailOrNull.prototype['active'] = false;
/**
 * A freeform descriptive note.
 * @member {String} comment
 */

ServiceVersionDetailOrNull.prototype['comment'] = undefined;
/**
 * Unused at this time.
 * @member {Boolean} deployed
 */

ServiceVersionDetailOrNull.prototype['deployed'] = undefined;
/**
 * Whether this version is locked or not. Objects can not be added or edited on locked versions.
 * @member {Boolean} locked
 * @default false
 */

ServiceVersionDetailOrNull.prototype['locked'] = false;
/**
 * The number of this version.
 * @member {Number} number
 */

ServiceVersionDetailOrNull.prototype['number'] = undefined;
/**
 * Unused at this time.
 * @member {Boolean} staging
 * @default false
 */

ServiceVersionDetailOrNull.prototype['staging'] = false;
/**
 * Unused at this time.
 * @member {Boolean} testing
 * @default false
 */

ServiceVersionDetailOrNull.prototype['testing'] = false;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */

ServiceVersionDetailOrNull.prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

ServiceVersionDetailOrNull.prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

ServiceVersionDetailOrNull.prototype['updated_at'] = undefined;
/**
 * @member {String} service_id
 */

ServiceVersionDetailOrNull.prototype['service_id'] = undefined;
/**
 * List of backends associated to this service.
 * @member {Array.<module:model/BackendResponse>} backends
 */

ServiceVersionDetailOrNull.prototype['backends'] = undefined;
/**
 * List of cache settings associated to this service.
 * @member {Array.<module:model/CacheSettingResponse>} cache_settings
 */

ServiceVersionDetailOrNull.prototype['cache_settings'] = undefined;
/**
 * List of conditions associated to this service.
 * @member {Array.<module:model/ConditionResponse>} conditions
 */

ServiceVersionDetailOrNull.prototype['conditions'] = undefined;
/**
 * List of directors associated to this service.
 * @member {Array.<module:model/Director>} directors
 */

ServiceVersionDetailOrNull.prototype['directors'] = undefined;
/**
 * List of domains associated to this service.
 * @member {Array.<module:model/DomainResponse>} domains
 */

ServiceVersionDetailOrNull.prototype['domains'] = undefined;
/**
 * List of gzip rules associated to this service.
 * @member {Array.<module:model/GzipResponse>} gzips
 */

ServiceVersionDetailOrNull.prototype['gzips'] = undefined;
/**
 * List of headers associated to this service.
 * @member {Array.<module:model/HeaderResponse>} headers
 */

ServiceVersionDetailOrNull.prototype['headers'] = undefined;
/**
 * List of healthchecks associated to this service.
 * @member {Array.<module:model/HealthcheckResponse>} healthchecks
 */

ServiceVersionDetailOrNull.prototype['healthchecks'] = undefined;
/**
 * List of request settings for this service.
 * @member {Array.<module:model/RequestSettingsResponse>} request_settings
 */

ServiceVersionDetailOrNull.prototype['request_settings'] = undefined;
/**
 * List of response objects for this service.
 * @member {Array.<module:model/ResponseObjectResponse>} response_objects
 */

ServiceVersionDetailOrNull.prototype['response_objects'] = undefined;
/**
 * List of default settings for this service.
 * @member {module:model/Settings} settings
 */

ServiceVersionDetailOrNull.prototype['settings'] = undefined;
/**
 * List of VCL snippets for this service.
 * @member {Array.<module:model/SchemasSnippetResponse>} snippets
 */

ServiceVersionDetailOrNull.prototype['snippets'] = undefined;
/**
 * List of VCL files for this service.
 * @member {Array.<module:model/SchemasVclResponse>} vcls
 */

ServiceVersionDetailOrNull.prototype['vcls'] = undefined;
/**
 * A list of Wordpress rules with this service.
 * @member {Array.<Object>} wordpress
 */

ServiceVersionDetailOrNull.prototype['wordpress'] = undefined; // Implement ServiceVersionDetail interface:

/**
 * Whether this is the active version or not.
 * @member {Boolean} active
 * @default false
 */

_ServiceVersionDetail["default"].prototype['active'] = false;
/**
 * A freeform descriptive note.
 * @member {String} comment
 */

_ServiceVersionDetail["default"].prototype['comment'] = undefined;
/**
 * Unused at this time.
 * @member {Boolean} deployed
 */

_ServiceVersionDetail["default"].prototype['deployed'] = undefined;
/**
 * Whether this version is locked or not. Objects can not be added or edited on locked versions.
 * @member {Boolean} locked
 * @default false
 */

_ServiceVersionDetail["default"].prototype['locked'] = false;
/**
 * The number of this version.
 * @member {Number} number
 */

_ServiceVersionDetail["default"].prototype['number'] = undefined;
/**
 * Unused at this time.
 * @member {Boolean} staging
 * @default false
 */

_ServiceVersionDetail["default"].prototype['staging'] = false;
/**
 * Unused at this time.
 * @member {Boolean} testing
 * @default false
 */

_ServiceVersionDetail["default"].prototype['testing'] = false;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */

_ServiceVersionDetail["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

_ServiceVersionDetail["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

_ServiceVersionDetail["default"].prototype['updated_at'] = undefined;
/**
 * @member {String} service_id
 */

_ServiceVersionDetail["default"].prototype['service_id'] = undefined;
/**
 * List of backends associated to this service.
 * @member {Array.<module:model/BackendResponse>} backends
 */

_ServiceVersionDetail["default"].prototype['backends'] = undefined;
/**
 * List of cache settings associated to this service.
 * @member {Array.<module:model/CacheSettingResponse>} cache_settings
 */

_ServiceVersionDetail["default"].prototype['cache_settings'] = undefined;
/**
 * List of conditions associated to this service.
 * @member {Array.<module:model/ConditionResponse>} conditions
 */

_ServiceVersionDetail["default"].prototype['conditions'] = undefined;
/**
 * List of directors associated to this service.
 * @member {Array.<module:model/Director>} directors
 */

_ServiceVersionDetail["default"].prototype['directors'] = undefined;
/**
 * List of domains associated to this service.
 * @member {Array.<module:model/DomainResponse>} domains
 */

_ServiceVersionDetail["default"].prototype['domains'] = undefined;
/**
 * List of gzip rules associated to this service.
 * @member {Array.<module:model/GzipResponse>} gzips
 */

_ServiceVersionDetail["default"].prototype['gzips'] = undefined;
/**
 * List of headers associated to this service.
 * @member {Array.<module:model/HeaderResponse>} headers
 */

_ServiceVersionDetail["default"].prototype['headers'] = undefined;
/**
 * List of healthchecks associated to this service.
 * @member {Array.<module:model/HealthcheckResponse>} healthchecks
 */

_ServiceVersionDetail["default"].prototype['healthchecks'] = undefined;
/**
 * List of request settings for this service.
 * @member {Array.<module:model/RequestSettingsResponse>} request_settings
 */

_ServiceVersionDetail["default"].prototype['request_settings'] = undefined;
/**
 * List of response objects for this service.
 * @member {Array.<module:model/ResponseObjectResponse>} response_objects
 */

_ServiceVersionDetail["default"].prototype['response_objects'] = undefined;
/**
 * List of default settings for this service.
 * @member {module:model/Settings} settings
 */

_ServiceVersionDetail["default"].prototype['settings'] = undefined;
/**
 * List of VCL snippets for this service.
 * @member {Array.<module:model/SchemasSnippetResponse>} snippets
 */

_ServiceVersionDetail["default"].prototype['snippets'] = undefined;
/**
 * List of VCL files for this service.
 * @member {Array.<module:model/SchemasVclResponse>} vcls
 */

_ServiceVersionDetail["default"].prototype['vcls'] = undefined;
/**
 * A list of Wordpress rules with this service.
 * @member {Array.<Object>} wordpress
 */

_ServiceVersionDetail["default"].prototype['wordpress'] = undefined;
var _default = ServiceVersionDetailOrNull;
exports["default"] = _default;
