// fastly/dist/model/TlsBulkCertificateResponseAttributesAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TlsBulkCertificateResponseAttributesAllOf model module.
 * @module model/TlsBulkCertificateResponseAttributesAllOf
 * @version 3.0.0-beta3
 */
var TlsBulkCertificateResponseAttributesAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TlsBulkCertificateResponseAttributesAllOf</code>.
   * @alias module:model/TlsBulkCertificateResponseAttributesAllOf
   */
  function TlsBulkCertificateResponseAttributesAllOf() {
    _classCallCheck(this, TlsBulkCertificateResponseAttributesAllOf);

    TlsBulkCertificateResponseAttributesAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TlsBulkCertificateResponseAttributesAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TlsBulkCertificateResponseAttributesAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TlsBulkCertificateResponseAttributesAllOf} obj Optional instance to populate.
     * @return {module:model/TlsBulkCertificateResponseAttributesAllOf} The populated <code>TlsBulkCertificateResponseAttributesAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TlsBulkCertificateResponseAttributesAllOf();

        if (data.hasOwnProperty('not_after')) {
          obj['not_after'] = _ApiClient["default"].convertToType(data['not_after'], 'Date');
        }

        if (data.hasOwnProperty('not_before')) {
          obj['not_before'] = _ApiClient["default"].convertToType(data['not_before'], 'Date');
        }

        if (data.hasOwnProperty('replace')) {
          obj['replace'] = _ApiClient["default"].convertToType(data['replace'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return TlsBulkCertificateResponseAttributesAllOf;
}();
/**
 * Time-stamp (GMT) when the certificate will expire. Must be in the future to be used to terminate TLS traffic.
 * @member {Date} not_after
 */


TlsBulkCertificateResponseAttributesAllOf.prototype['not_after'] = undefined;
/**
 * Time-stamp (GMT) when the certificate will become valid. Must be in the past to be used to terminate TLS traffic.
 * @member {Date} not_before
 */

TlsBulkCertificateResponseAttributesAllOf.prototype['not_before'] = undefined;
/**
 * A recommendation from Fastly indicating the key associated with this certificate is in need of rotation.
 * @member {Boolean} replace
 */

TlsBulkCertificateResponseAttributesAllOf.prototype['replace'] = undefined;
var _default = TlsBulkCertificateResponseAttributesAllOf;
exports["default"] = _default;
