// fastly/dist/model/TlsConfigurationResponseAttributes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Timestamps = _interopRequireDefault(require("./Timestamps"));

var _TlsConfigurationResponseAttributesAllOf = _interopRequireDefault(require("./TlsConfigurationResponseAttributesAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TlsConfigurationResponseAttributes model module.
 * @module model/TlsConfigurationResponseAttributes
 * @version 3.0.0-beta3
 */
var TlsConfigurationResponseAttributes = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TlsConfigurationResponseAttributes</code>.
   * @alias module:model/TlsConfigurationResponseAttributes
   * @implements module:model/Timestamps
   * @implements module:model/TlsConfigurationResponseAttributesAllOf
   */
  function TlsConfigurationResponseAttributes() {
    _classCallCheck(this, TlsConfigurationResponseAttributes);

    _Timestamps["default"].initialize(this);

    _TlsConfigurationResponseAttributesAllOf["default"].initialize(this);

    TlsConfigurationResponseAttributes.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TlsConfigurationResponseAttributes, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TlsConfigurationResponseAttributes</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TlsConfigurationResponseAttributes} obj Optional instance to populate.
     * @return {module:model/TlsConfigurationResponseAttributes} The populated <code>TlsConfigurationResponseAttributes</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TlsConfigurationResponseAttributes();

        _Timestamps["default"].constructFromObject(data, obj);

        _TlsConfigurationResponseAttributesAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }

        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }

        if (data.hasOwnProperty('default')) {
          obj['default'] = _ApiClient["default"].convertToType(data['default'], 'Boolean');
        }

        if (data.hasOwnProperty('http_protocols')) {
          obj['http_protocols'] = _ApiClient["default"].convertToType(data['http_protocols'], ['String']);
        }

        if (data.hasOwnProperty('tls_protocols')) {
          obj['tls_protocols'] = _ApiClient["default"].convertToType(data['tls_protocols'], ['Number']);
        }

        if (data.hasOwnProperty('bulk')) {
          obj['bulk'] = _ApiClient["default"].convertToType(data['bulk'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return TlsConfigurationResponseAttributes;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */


TlsConfigurationResponseAttributes.prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

TlsConfigurationResponseAttributes.prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

TlsConfigurationResponseAttributes.prototype['updated_at'] = undefined;
/**
 * Signifies whether or not Fastly will use this configuration as a default when creating a new [TLS Activation](/reference/api/tls/custom-certs/activations/).
 * @member {Boolean} default
 */

TlsConfigurationResponseAttributes.prototype['default'] = undefined;
/**
 * HTTP protocols available on your configuration.
 * @member {Array.<String>} http_protocols
 */

TlsConfigurationResponseAttributes.prototype['http_protocols'] = undefined;
/**
 * TLS protocols available on your configuration.
 * @member {Array.<Number>} tls_protocols
 */

TlsConfigurationResponseAttributes.prototype['tls_protocols'] = undefined;
/**
 * Signifies whether the configuration is used for Platform TLS or not.
 * @member {Boolean} bulk
 */

TlsConfigurationResponseAttributes.prototype['bulk'] = undefined; // Implement Timestamps interface:

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */

_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

_Timestamps["default"].prototype['updated_at'] = undefined; // Implement TlsConfigurationResponseAttributesAllOf interface:

/**
 * Signifies whether or not Fastly will use this configuration as a default when creating a new [TLS Activation](/reference/api/tls/custom-certs/activations/).
 * @member {Boolean} default
 */

_TlsConfigurationResponseAttributesAllOf["default"].prototype['default'] = undefined;
/**
 * HTTP protocols available on your configuration.
 * @member {Array.<String>} http_protocols
 */

_TlsConfigurationResponseAttributesAllOf["default"].prototype['http_protocols'] = undefined;
/**
 * TLS protocols available on your configuration.
 * @member {Array.<Number>} tls_protocols
 */

_TlsConfigurationResponseAttributesAllOf["default"].prototype['tls_protocols'] = undefined;
/**
 * Signifies whether the configuration is used for Platform TLS or not.
 * @member {Boolean} bulk
 */

_TlsConfigurationResponseAttributesAllOf["default"].prototype['bulk'] = undefined;
var _default = TlsConfigurationResponseAttributes;
exports["default"] = _default;
