// fastly/dist/model/TlsCertificateResponseAttributesAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TlsCertificateResponseAttributesAllOf model module.
 * @module model/TlsCertificateResponseAttributesAllOf
 * @version 3.0.0-beta3
 */
var TlsCertificateResponseAttributesAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TlsCertificateResponseAttributesAllOf</code>.
   * @alias module:model/TlsCertificateResponseAttributesAllOf
   */
  function TlsCertificateResponseAttributesAllOf() {
    _classCallCheck(this, TlsCertificateResponseAttributesAllOf);

    TlsCertificateResponseAttributesAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TlsCertificateResponseAttributesAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TlsCertificateResponseAttributesAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TlsCertificateResponseAttributesAllOf} obj Optional instance to populate.
     * @return {module:model/TlsCertificateResponseAttributesAllOf} The populated <code>TlsCertificateResponseAttributesAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TlsCertificateResponseAttributesAllOf();

        if (data.hasOwnProperty('issued_to')) {
          obj['issued_to'] = _ApiClient["default"].convertToType(data['issued_to'], 'String');
        }

        if (data.hasOwnProperty('issuer')) {
          obj['issuer'] = _ApiClient["default"].convertToType(data['issuer'], 'String');
        }

        if (data.hasOwnProperty('serial_number')) {
          obj['serial_number'] = _ApiClient["default"].convertToType(data['serial_number'], 'String');
        }

        if (data.hasOwnProperty('signature_algorithm')) {
          obj['signature_algorithm'] = _ApiClient["default"].convertToType(data['signature_algorithm'], 'String');
        }

        if (data.hasOwnProperty('not_after')) {
          obj['not_after'] = _ApiClient["default"].convertToType(data['not_after'], 'Date');
        }

        if (data.hasOwnProperty('not_before')) {
          obj['not_before'] = _ApiClient["default"].convertToType(data['not_before'], 'Date');
        }

        if (data.hasOwnProperty('replace')) {
          obj['replace'] = _ApiClient["default"].convertToType(data['replace'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return TlsCertificateResponseAttributesAllOf;
}();
/**
 * The hostname for which a certificate was issued.
 * @member {String} issued_to
 */


TlsCertificateResponseAttributesAllOf.prototype['issued_to'] = undefined;
/**
 * The certificate authority that issued the certificate.
 * @member {String} issuer
 */

TlsCertificateResponseAttributesAllOf.prototype['issuer'] = undefined;
/**
 * A value assigned by the issuer that is unique to a certificate.
 * @member {String} serial_number
 */

TlsCertificateResponseAttributesAllOf.prototype['serial_number'] = undefined;
/**
 * The algorithm used to sign the certificate.
 * @member {String} signature_algorithm
 */

TlsCertificateResponseAttributesAllOf.prototype['signature_algorithm'] = undefined;
/**
 * Time-stamp (GMT) when the certificate will expire. Must be in the future to be used to terminate TLS traffic.
 * @member {Date} not_after
 */

TlsCertificateResponseAttributesAllOf.prototype['not_after'] = undefined;
/**
 * Time-stamp (GMT) when the certificate will become valid. Must be in the past to be used to terminate TLS traffic.
 * @member {Date} not_before
 */

TlsCertificateResponseAttributesAllOf.prototype['not_before'] = undefined;
/**
 * A recommendation from Fastly indicating the key associated with this certificate is in need of rotation.
 * @member {Boolean} replace
 */

TlsCertificateResponseAttributesAllOf.prototype['replace'] = undefined;
var _default = TlsCertificateResponseAttributesAllOf;
exports["default"] = _default;
