// fastly/dist/model/TlsConfigurationResponseAttributesAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TlsConfigurationResponseAttributesAllOf model module.
 * @module model/TlsConfigurationResponseAttributesAllOf
 * @version 3.0.0-beta3
 */
var TlsConfigurationResponseAttributesAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TlsConfigurationResponseAttributesAllOf</code>.
   * @alias module:model/TlsConfigurationResponseAttributesAllOf
   */
  function TlsConfigurationResponseAttributesAllOf() {
    _classCallCheck(this, TlsConfigurationResponseAttributesAllOf);

    TlsConfigurationResponseAttributesAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TlsConfigurationResponseAttributesAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TlsConfigurationResponseAttributesAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TlsConfigurationResponseAttributesAllOf} obj Optional instance to populate.
     * @return {module:model/TlsConfigurationResponseAttributesAllOf} The populated <code>TlsConfigurationResponseAttributesAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TlsConfigurationResponseAttributesAllOf();

        if (data.hasOwnProperty('default')) {
          obj['default'] = _ApiClient["default"].convertToType(data['default'], 'Boolean');
        }

        if (data.hasOwnProperty('http_protocols')) {
          obj['http_protocols'] = _ApiClient["default"].convertToType(data['http_protocols'], ['String']);
        }

        if (data.hasOwnProperty('tls_protocols')) {
          obj['tls_protocols'] = _ApiClient["default"].convertToType(data['tls_protocols'], ['Number']);
        }

        if (data.hasOwnProperty('bulk')) {
          obj['bulk'] = _ApiClient["default"].convertToType(data['bulk'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return TlsConfigurationResponseAttributesAllOf;
}();
/**
 * Signifies whether or not Fastly will use this configuration as a default when creating a new [TLS Activation](/reference/api/tls/custom-certs/activations/).
 * @member {Boolean} default
 */


TlsConfigurationResponseAttributesAllOf.prototype['default'] = undefined;
/**
 * HTTP protocols available on your configuration.
 * @member {Array.<String>} http_protocols
 */

TlsConfigurationResponseAttributesAllOf.prototype['http_protocols'] = undefined;
/**
 * TLS protocols available on your configuration.
 * @member {Array.<Number>} tls_protocols
 */

TlsConfigurationResponseAttributesAllOf.prototype['tls_protocols'] = undefined;
/**
 * Signifies whether the configuration is used for Platform TLS or not.
 * @member {Boolean} bulk
 */

TlsConfigurationResponseAttributesAllOf.prototype['bulk'] = undefined;
var _default = TlsConfigurationResponseAttributesAllOf;
exports["default"] = _default;
