// fastly/dist/model/TlsSubscriptionResponseAttributes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Timestamps = _interopRequireDefault(require("./Timestamps"));

var _TlsSubscriptionResponseAttributesAllOf = _interopRequireDefault(require("./TlsSubscriptionResponseAttributesAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TlsSubscriptionResponseAttributes model module.
 * @module model/TlsSubscriptionResponseAttributes
 * @version 3.0.0-beta3
 */
var TlsSubscriptionResponseAttributes = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TlsSubscriptionResponseAttributes</code>.
   * @alias module:model/TlsSubscriptionResponseAttributes
   * @implements module:model/Timestamps
   * @implements module:model/TlsSubscriptionResponseAttributesAllOf
   */
  function TlsSubscriptionResponseAttributes() {
    _classCallCheck(this, TlsSubscriptionResponseAttributes);

    _Timestamps["default"].initialize(this);

    _TlsSubscriptionResponseAttributesAllOf["default"].initialize(this);

    TlsSubscriptionResponseAttributes.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TlsSubscriptionResponseAttributes, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TlsSubscriptionResponseAttributes</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TlsSubscriptionResponseAttributes} obj Optional instance to populate.
     * @return {module:model/TlsSubscriptionResponseAttributes} The populated <code>TlsSubscriptionResponseAttributes</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TlsSubscriptionResponseAttributes();

        _Timestamps["default"].constructFromObject(data, obj);

        _TlsSubscriptionResponseAttributesAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }

        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }

        if (data.hasOwnProperty('state')) {
          obj['state'] = _ApiClient["default"].convertToType(data['state'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TlsSubscriptionResponseAttributes;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */


TlsSubscriptionResponseAttributes.prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

TlsSubscriptionResponseAttributes.prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

TlsSubscriptionResponseAttributes.prototype['updated_at'] = undefined;
/**
 * The current state of your subscription.
 * @member {module:model/TlsSubscriptionResponseAttributes.StateEnum} state
 */

TlsSubscriptionResponseAttributes.prototype['state'] = undefined; // Implement Timestamps interface:

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */

_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

_Timestamps["default"].prototype['updated_at'] = undefined; // Implement TlsSubscriptionResponseAttributesAllOf interface:

/**
 * The current state of your subscription.
 * @member {module:model/TlsSubscriptionResponseAttributesAllOf.StateEnum} state
 */

_TlsSubscriptionResponseAttributesAllOf["default"].prototype['state'] = undefined;
/**
 * Allowed values for the <code>state</code> property.
 * @enum {String}
 * @readonly
 */

TlsSubscriptionResponseAttributes['StateEnum'] = {
  /**
   * value: "pending"
   * @const
   */
  "pending": "pending",

  /**
   * value: "processing"
   * @const
   */
  "processing": "processing",

  /**
   * value: "issued"
   * @const
   */
  "issued": "issued",

  /**
   * value: "renewing"
   * @const
   */
  "renewing": "renewing"
};
var _default = TlsSubscriptionResponseAttributes;
exports["default"] = _default;
