// fastly/dist/model/WafFirewallResponseDataAttributes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Timestamps = _interopRequireDefault(require("./Timestamps"));

var _WafFirewallResponseDataAttributesAllOf = _interopRequireDefault(require("./WafFirewallResponseDataAttributesAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The WafFirewallResponseDataAttributes model module.
 * @module model/WafFirewallResponseDataAttributes
 * @version 3.0.0-beta3
 */
var WafFirewallResponseDataAttributes = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WafFirewallResponseDataAttributes</code>.
   * @alias module:model/WafFirewallResponseDataAttributes
   * @implements module:model/Timestamps
   * @implements module:model/WafFirewallResponseDataAttributesAllOf
   */
  function WafFirewallResponseDataAttributes() {
    _classCallCheck(this, WafFirewallResponseDataAttributes);

    _Timestamps["default"].initialize(this);

    _WafFirewallResponseDataAttributesAllOf["default"].initialize(this);

    WafFirewallResponseDataAttributes.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(WafFirewallResponseDataAttributes, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WafFirewallResponseDataAttributes</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WafFirewallResponseDataAttributes} obj Optional instance to populate.
     * @return {module:model/WafFirewallResponseDataAttributes} The populated <code>WafFirewallResponseDataAttributes</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WafFirewallResponseDataAttributes();

        _Timestamps["default"].constructFromObject(data, obj);

        _WafFirewallResponseDataAttributesAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }

        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }

        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }

        if (data.hasOwnProperty('active_rules_fastly_block_count')) {
          obj['active_rules_fastly_block_count'] = _ApiClient["default"].convertToType(data['active_rules_fastly_block_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_fastly_log_count')) {
          obj['active_rules_fastly_log_count'] = _ApiClient["default"].convertToType(data['active_rules_fastly_log_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_fastly_score_count')) {
          obj['active_rules_fastly_score_count'] = _ApiClient["default"].convertToType(data['active_rules_fastly_score_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_owasp_block_count')) {
          obj['active_rules_owasp_block_count'] = _ApiClient["default"].convertToType(data['active_rules_owasp_block_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_owasp_log_count')) {
          obj['active_rules_owasp_log_count'] = _ApiClient["default"].convertToType(data['active_rules_owasp_log_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_owasp_score_count')) {
          obj['active_rules_owasp_score_count'] = _ApiClient["default"].convertToType(data['active_rules_owasp_score_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_trustwave_block_count')) {
          obj['active_rules_trustwave_block_count'] = _ApiClient["default"].convertToType(data['active_rules_trustwave_block_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_trustwave_log_count')) {
          obj['active_rules_trustwave_log_count'] = _ApiClient["default"].convertToType(data['active_rules_trustwave_log_count'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return WafFirewallResponseDataAttributes;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */


WafFirewallResponseDataAttributes.prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

WafFirewallResponseDataAttributes.prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

WafFirewallResponseDataAttributes.prototype['updated_at'] = undefined;
/**
 * @member {String} service_id
 */

WafFirewallResponseDataAttributes.prototype['service_id'] = undefined;
/**
 * The number of active Fastly rules set to block on the active or latest firewall version.
 * @member {Number} active_rules_fastly_block_count
 */

WafFirewallResponseDataAttributes.prototype['active_rules_fastly_block_count'] = undefined;
/**
 * The number of active Fastly rules set to log on the active or latest firewall version.
 * @member {Number} active_rules_fastly_log_count
 */

WafFirewallResponseDataAttributes.prototype['active_rules_fastly_log_count'] = undefined;
/**
 * The number of active Fastly rules set to score on the active or latest firewall version.
 * @member {Number} active_rules_fastly_score_count
 */

WafFirewallResponseDataAttributes.prototype['active_rules_fastly_score_count'] = undefined;
/**
 * The number of active OWASP rules set to block on the active or latest firewall version.
 * @member {Number} active_rules_owasp_block_count
 */

WafFirewallResponseDataAttributes.prototype['active_rules_owasp_block_count'] = undefined;
/**
 * The number of active OWASP rules set to log on the active or latest firewall version.
 * @member {Number} active_rules_owasp_log_count
 */

WafFirewallResponseDataAttributes.prototype['active_rules_owasp_log_count'] = undefined;
/**
 * The number of active OWASP rules set to score on the active or latest firewall version.
 * @member {Number} active_rules_owasp_score_count
 */

WafFirewallResponseDataAttributes.prototype['active_rules_owasp_score_count'] = undefined;
/**
 * The number of active Trustwave rules set to block on the active or latest firewall version.
 * @member {Number} active_rules_trustwave_block_count
 */

WafFirewallResponseDataAttributes.prototype['active_rules_trustwave_block_count'] = undefined;
/**
 * The number of active Trustwave rules set to log on the active or latest firewall version.
 * @member {Number} active_rules_trustwave_log_count
 */

WafFirewallResponseDataAttributes.prototype['active_rules_trustwave_log_count'] = undefined; // Implement Timestamps interface:

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */

_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */

_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */

_Timestamps["default"].prototype['updated_at'] = undefined; // Implement WafFirewallResponseDataAttributesAllOf interface:

/**
 * @member {String} service_id
 */

_WafFirewallResponseDataAttributesAllOf["default"].prototype['service_id'] = undefined;
/**
 * The number of active Fastly rules set to block on the active or latest firewall version.
 * @member {Number} active_rules_fastly_block_count
 */

_WafFirewallResponseDataAttributesAllOf["default"].prototype['active_rules_fastly_block_count'] = undefined;
/**
 * The number of active Fastly rules set to log on the active or latest firewall version.
 * @member {Number} active_rules_fastly_log_count
 */

_WafFirewallResponseDataAttributesAllOf["default"].prototype['active_rules_fastly_log_count'] = undefined;
/**
 * The number of active Fastly rules set to score on the active or latest firewall version.
 * @member {Number} active_rules_fastly_score_count
 */

_WafFirewallResponseDataAttributesAllOf["default"].prototype['active_rules_fastly_score_count'] = undefined;
/**
 * The number of active OWASP rules set to block on the active or latest firewall version.
 * @member {Number} active_rules_owasp_block_count
 */

_WafFirewallResponseDataAttributesAllOf["default"].prototype['active_rules_owasp_block_count'] = undefined;
/**
 * The number of active OWASP rules set to log on the active or latest firewall version.
 * @member {Number} active_rules_owasp_log_count
 */

_WafFirewallResponseDataAttributesAllOf["default"].prototype['active_rules_owasp_log_count'] = undefined;
/**
 * The number of active OWASP rules set to score on the active or latest firewall version.
 * @member {Number} active_rules_owasp_score_count
 */

_WafFirewallResponseDataAttributesAllOf["default"].prototype['active_rules_owasp_score_count'] = undefined;
/**
 * The number of active Trustwave rules set to block on the active or latest firewall version.
 * @member {Number} active_rules_trustwave_block_count
 */

_WafFirewallResponseDataAttributesAllOf["default"].prototype['active_rules_trustwave_block_count'] = undefined;
/**
 * The number of active Trustwave rules set to log on the active or latest firewall version.
 * @member {Number} active_rules_trustwave_log_count
 */

_WafFirewallResponseDataAttributesAllOf["default"].prototype['active_rules_trustwave_log_count'] = undefined;
var _default = WafFirewallResponseDataAttributes;
exports["default"] = _default;
