// fastly/dist/model/WafFirewallResponseDataAttributesAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The WafFirewallResponseDataAttributesAllOf model module.
 * @module model/WafFirewallResponseDataAttributesAllOf
 * @version 3.0.0-beta3
 */
var WafFirewallResponseDataAttributesAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WafFirewallResponseDataAttributesAllOf</code>.
   * @alias module:model/WafFirewallResponseDataAttributesAllOf
   */
  function WafFirewallResponseDataAttributesAllOf() {
    _classCallCheck(this, WafFirewallResponseDataAttributesAllOf);

    WafFirewallResponseDataAttributesAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(WafFirewallResponseDataAttributesAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WafFirewallResponseDataAttributesAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WafFirewallResponseDataAttributesAllOf} obj Optional instance to populate.
     * @return {module:model/WafFirewallResponseDataAttributesAllOf} The populated <code>WafFirewallResponseDataAttributesAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WafFirewallResponseDataAttributesAllOf();

        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }

        if (data.hasOwnProperty('active_rules_fastly_block_count')) {
          obj['active_rules_fastly_block_count'] = _ApiClient["default"].convertToType(data['active_rules_fastly_block_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_fastly_log_count')) {
          obj['active_rules_fastly_log_count'] = _ApiClient["default"].convertToType(data['active_rules_fastly_log_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_fastly_score_count')) {
          obj['active_rules_fastly_score_count'] = _ApiClient["default"].convertToType(data['active_rules_fastly_score_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_owasp_block_count')) {
          obj['active_rules_owasp_block_count'] = _ApiClient["default"].convertToType(data['active_rules_owasp_block_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_owasp_log_count')) {
          obj['active_rules_owasp_log_count'] = _ApiClient["default"].convertToType(data['active_rules_owasp_log_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_owasp_score_count')) {
          obj['active_rules_owasp_score_count'] = _ApiClient["default"].convertToType(data['active_rules_owasp_score_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_trustwave_block_count')) {
          obj['active_rules_trustwave_block_count'] = _ApiClient["default"].convertToType(data['active_rules_trustwave_block_count'], 'Number');
        }

        if (data.hasOwnProperty('active_rules_trustwave_log_count')) {
          obj['active_rules_trustwave_log_count'] = _ApiClient["default"].convertToType(data['active_rules_trustwave_log_count'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return WafFirewallResponseDataAttributesAllOf;
}();
/**
 * @member {String} service_id
 */


WafFirewallResponseDataAttributesAllOf.prototype['service_id'] = undefined;
/**
 * The number of active Fastly rules set to block on the active or latest firewall version.
 * @member {Number} active_rules_fastly_block_count
 */

WafFirewallResponseDataAttributesAllOf.prototype['active_rules_fastly_block_count'] = undefined;
/**
 * The number of active Fastly rules set to log on the active or latest firewall version.
 * @member {Number} active_rules_fastly_log_count
 */

WafFirewallResponseDataAttributesAllOf.prototype['active_rules_fastly_log_count'] = undefined;
/**
 * The number of active Fastly rules set to score on the active or latest firewall version.
 * @member {Number} active_rules_fastly_score_count
 */

WafFirewallResponseDataAttributesAllOf.prototype['active_rules_fastly_score_count'] = undefined;
/**
 * The number of active OWASP rules set to block on the active or latest firewall version.
 * @member {Number} active_rules_owasp_block_count
 */

WafFirewallResponseDataAttributesAllOf.prototype['active_rules_owasp_block_count'] = undefined;
/**
 * The number of active OWASP rules set to log on the active or latest firewall version.
 * @member {Number} active_rules_owasp_log_count
 */

WafFirewallResponseDataAttributesAllOf.prototype['active_rules_owasp_log_count'] = undefined;
/**
 * The number of active OWASP rules set to score on the active or latest firewall version.
 * @member {Number} active_rules_owasp_score_count
 */

WafFirewallResponseDataAttributesAllOf.prototype['active_rules_owasp_score_count'] = undefined;
/**
 * The number of active Trustwave rules set to block on the active or latest firewall version.
 * @member {Number} active_rules_trustwave_block_count
 */

WafFirewallResponseDataAttributesAllOf.prototype['active_rules_trustwave_block_count'] = undefined;
/**
 * The number of active Trustwave rules set to log on the active or latest firewall version.
 * @member {Number} active_rules_trustwave_log_count
 */

WafFirewallResponseDataAttributesAllOf.prototype['active_rules_trustwave_log_count'] = undefined;
var _default = WafFirewallResponseDataAttributesAllOf;
exports["default"] = _default;
