// fastly/dist/model/WafTagsResponseDataItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RelationshipWafRule = _interopRequireDefault(require("./RelationshipWafRule"));

var _TypeWafTag = _interopRequireDefault(require("./TypeWafTag"));

var _WafRuleRevisionResponseDataAllOf = _interopRequireDefault(require("./WafRuleRevisionResponseDataAllOf"));

var _WafTag = _interopRequireDefault(require("./WafTag"));

var _WafTagAttributes = _interopRequireDefault(require("./WafTagAttributes"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The WafTagsResponseDataItem model module.
 * @module model/WafTagsResponseDataItem
 * @version 3.0.0-beta3
 */
var WafTagsResponseDataItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WafTagsResponseDataItem</code>.
   * @alias module:model/WafTagsResponseDataItem
   * @implements module:model/WafTag
   * @implements module:model/WafRuleRevisionResponseDataAllOf
   */
  function WafTagsResponseDataItem() {
    _classCallCheck(this, WafTagsResponseDataItem);

    _WafTag["default"].initialize(this);

    _WafRuleRevisionResponseDataAllOf["default"].initialize(this);

    WafTagsResponseDataItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(WafTagsResponseDataItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WafTagsResponseDataItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WafTagsResponseDataItem} obj Optional instance to populate.
     * @return {module:model/WafTagsResponseDataItem} The populated <code>WafTagsResponseDataItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WafTagsResponseDataItem();

        _WafTag["default"].constructFromObject(data, obj);

        _WafRuleRevisionResponseDataAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('type')) {
          obj['type'] = _TypeWafTag["default"].constructFromObject(data['type']);
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('attributes')) {
          obj['attributes'] = _WafTagAttributes["default"].constructFromObject(data['attributes']);
        }

        if (data.hasOwnProperty('relationships')) {
          obj['relationships'] = _RelationshipWafRule["default"].constructFromObject(data['relationships']);
        }
      }

      return obj;
    }
  }]);

  return WafTagsResponseDataItem;
}();
/**
 * @member {module:model/TypeWafTag} type
 */


WafTagsResponseDataItem.prototype['type'] = undefined;
/**
 * Alphanumeric string identifying a WAF tag.
 * @member {String} id
 */

WafTagsResponseDataItem.prototype['id'] = undefined;
/**
 * @member {module:model/WafTagAttributes} attributes
 */

WafTagsResponseDataItem.prototype['attributes'] = undefined;
/**
 * @member {module:model/RelationshipWafRule} relationships
 */

WafTagsResponseDataItem.prototype['relationships'] = undefined; // Implement WafTag interface:

/**
 * @member {module:model/TypeWafTag} type
 */

_WafTag["default"].prototype['type'] = undefined;
/**
 * Alphanumeric string identifying a WAF tag.
 * @member {String} id
 */

_WafTag["default"].prototype['id'] = undefined;
/**
 * @member {module:model/WafTagAttributes} attributes
 */

_WafTag["default"].prototype['attributes'] = undefined; // Implement WafRuleRevisionResponseDataAllOf interface:

/**
 * @member {module:model/RelationshipWafRule} relationships
 */

_WafRuleRevisionResponseDataAllOf["default"].prototype['relationships'] = undefined;
var _default = WafTagsResponseDataItem;
exports["default"] = _default;
